kura/org.eclipse.kura.ai.triton.server/src/main/java/org/eclipse/kura/ai/triton/server/TritonServerServiceImpl.java

```diff
@@ -151,6 +153,16 @@ private void startLocalInstance() {
 
     private void stopLocalInstance() {
         this.tritonServerLocalManager.stop();
+
+        int counter = 0;
+        while (this.tritonServerLocalManager.isLocalServerRunning()) {
+            if (counter++ >= N_ATTEMPT) {
+                logger.warn("Cannot stop local server instance.");
+                return;
+            }
+            TritonServerLocalManager.sleepFor(WAIT_ATTEMPT_MS);
+        }
+
```

Comment:
Indeed the N_ATTEMPTS and WAIT_ATTEMPTS_MS seems to be a bit arbitrary... even if this is an improvement compared with the previous behavior. Is there a way to improve this further? Maybe spawing a thread for checking the server status and calling a callback upon termination?

Reply:
I agree. I already had a discussion about this with @MMaiero and we were opting for exposing these two parameters in the metatype for the user to choose.

I also like the callback solution but its scope may require a new PR.

Which of the two would you prefer?

Reply:
Good for me. If you agree I would open a new PR for that since it's a meaningful change in the UX that exceeds the scope of this PR (startup bugfix).